Src/Common/HeadersUtilities.cs

```diff
@@ -0,0 +1,139 @@
+                    }
+                    else
+                    {
+                        string name = TrimSubstring(keyValuePair, 0, equalsSignIndex);
+                        if (name == null)
+                        {
+                            Append(result, keyValuePair.Trim());
+                        }
+                        else if (name != keyName)
+                        {
+                            string value = TrimSubstring(keyValuePair, equalsSignIndex + 1, keyValuePair.Length);
+                            if (value == null)
+                            {
+                                value = string.Empty;
+                            }
+                            Append(result, CreateKeyValuePair(name, value));
+                        }
+                        else if (!found)
+                        {
+                            found = true;
+                            Append(result, CreateKeyValuePair(keyName.Trim(), keyValue.Trim()));
+                        }
+                    }
+                }
+            }
+
+            if (!found)
+            {
+                Append(result, CreateKeyValuePair(keyName.Trim(), keyValue.Trim()));
+            }
+
+            return result.ToString();
+        }
+
+        /// <summary>
+        /// Append the provided toAppend string after the provided value string.
+        /// </summary>
+        /// <param name="value">The string to append toAppend to.</param>
+        /// <param name="toAppend">The string to append after value.</param>
+        private static void Append(StringBuilder value, string toAppend)
+        {
+            if (value.Length > 0)
+            {
+                value.Append(", ");
+            }
+            value.Append(toAppend);
+        }
+
+        private static string TrimSubstring(string value, int startIndex, int endIndex)
+        {
+            int firstNonWhitespaceIndex = -1;
+            int last = -1;
+            for (int firstSearchIndex = startIndex; firstSearchIndex < endIndex; ++firstSearchIndex)
+            {
+                if (!char.IsWhiteSpace(value[firstSearchIndex]))
+                {
+                    firstNonWhitespaceIndex = firstSearchIndex;
+                    
+                    // Found the first non-whitespace character index, now look for the last.
+                    for (int lastSearchIndex = endIndex - 1; lastSearchIndex >= startIndex; --lastSearchIndex)
+                    {
+                        if (!char.IsWhiteSpace(value[lastSearchIndex]))
+                        {
+                            last = lastSearchIndex;
+                            break;
+                        }
+                    }
+                    break;
+                }
+            }
+
+            return firstNonWhitespaceIndex == -1 ? null : value.Substring(firstNonWhitespaceIndex, last - firstNonWhitespaceIndex + 1);
+        }
```

Comment:
No worries, I realized later that the checks would get more complicated to do what I was thinking and not simpler as intended.